lib/filter-stream.js

```diff
@@ -176,7 +177,7 @@ function FilterStream( blockMap, options ) {
     var buffer = Buffer.concat( this._buffers )
     var offset = 0
 
-    while( offset < buffer.length ) {
+    while( buffer.length - offset >= this.blockSize ) {
```

Comment:
I'm not sure if this is handled elsewhere, if its even a possible scenario, but could `buffer` not be a multiple of `this.blockSize`? For example, if `buffer` has size 16 and `this.blockSize` is 12 then:

- `buffer.length (16) - offset (0) >= this.blockSize (12)`, so this is true
- Push block  `buffer.slice( offset (0), offset (0) + this.blockSize (12) )`,
- `offset` is finally 12
- So `buffer.length (16) - offset (12) >= this.blockSize (12)`, false
- Are we ignoring the last 4 bytes of `buffer`?